tokio-current-thread/src/lib.rs

```diff
@@ -102,6 +102,32 @@ pub struct Entered<'a, P: Park + 'a> {
     enter: &'a mut Enter,
 }
 
+/// Future returned by the [`spawn_handle`] function.
+///
+/// A `SpawnHandle` will finish with either the `Item` type of the spawned
+/// future or a [`SpawnHandleError`] which either contains the spawned future's
+/// `Error` type or indicates that the spawned future was canceled before it
+/// completed.
+///
+/// In addition, the [`cancel`] method will cancel the spawned future, causing
+/// it to finish executing the next time it is polled.
+///
+/// [`spawn_handle`]: fn.spawn_handle.html
+/// [`SpawnHandleError`]: struct.SpawnHandleError.html
+/// [`cancel`]: #method.cancel
+#[derive(Debug)]
+pub struct SpawnHandle<T, E> {
+    cancel_tx: oneshot::Sender<()>,
```

Comment:
a cancellation oneshot should not be needed. You can use [this](https://docs.rs/futures/0.2.1/futures/channel/oneshot/struct.Sender.html#method.poll_cancel) fn instead.